syft/format/spdxtagvalue/decoder.go

```diff
@@ -0,0 +1,109 @@
+package spdxtagvalue
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spdx/tools-golang/tagvalue"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/format/common/spdxhelpers"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+var _ sbom.FormatDecoder = (*decoder)(nil)
+
+type decoder struct {
+}
+
+func NewFormatDecoder() sbom.FormatDecoder {
+	return decoder{}
+}
+
+func (d decoder) Decode(reader io.ReadSeeker) (*sbom.SBOM, sbom.FormatID, string, error) {
+	if reader == nil {
+		return nil, "", "", fmt.Errorf("no SBOM bytes provided")
+	}
+
+	// since spdx lib will always return the latest version of the document, we need to identify the version
+	// first and then decode into the appropriate document object. Otherwise if we get the version info from the
+	// decoded object we will always get the latest version (instead of the version we decoded from).
+	id, version := d.Identify(reader)
+	if id != ID {
+		return nil, "", "", fmt.Errorf("not a spdx tag-value document")
+	}
+	if version == "" {
+		return nil, "", "", fmt.Errorf("unsupported spdx tag-value document version")
+	}
+
+	if _, err := reader.Seek(0, io.SeekStart); err != nil {
+		return nil, "", "", fmt.Errorf("unable to seek to start of SPDX Tag-Value SBOM: %+v", err)
```

Comment:
nit: in the case a read seeker fails, it could log a debug statement and try to read anyway (same comment for other similar behavior)

Reply:
if seeking didn't work I think I'd prefer to stop as something is wrong with the underlying mechanics of the reader